test/cli/credentials/list.test.js

```diff
@@ -1,58 +1,187 @@
+      });
+  };
+
   before(() => {
-    ({ program, env } = environment.bootstrap());
+    ({program, env} = environment.bootstrap());
     return adminHelper.start();
   });
+
   after(() => adminHelper.stop());
 
   beforeEach(() => {
+    createdTypes.reset();
+    createdKeyAuthKeys.reset();
+
     env.prepareHijack();
-    return adminHelper.admin.users.create({
-      username: idGen.v4(),
-      firstname: 'La',
-      lastname: 'Deeda'
-    })
-    .then(createdUser => {
-      user = createdUser;
-      return Promise.all([
-        adminHelper.admin.credentials.create(user.username, 'key-auth', {}),
-        adminHelper.admin.credentials.create(user.username, 'basic-auth', {password: 'test'}),
-        adminHelper.admin.credentials.create(user.username, 'oauth2', {secret: 'eg'})
-      ]);
-    })
-    .then(([keyCred1Res, keyCred2Res, basicCredRes, oauth2CredRes]) => {
-      keyCred1 = keyCred1Res;
+
+    return adminHelper
+      .admin
+      .users
+      .create({
+        username: idGen.v4(),
+        firstname: 'La',
+        lastname: 'Deeda'
+      })
+      .then(user => {
+        username = user.username;
+        return Promise.all([
+          createCredential('key-auth'),
+          createCredential('basic-auth', {password: 'test1'}),
+          createCredential('oauth2', {secret: 'eg1'}),
+          createCredential('key-auth', {}, false),
+          createCredential('key-auth', {}, false),
+          createCredential('key-auth', {}, false)
+        ]);
+      });
+  });
+
+  it('should show active credentials', done => {
+    const types = {};
+    const keyAuthKeys = [];
+    env.hijack(namespace, generator => {
+      generator.once('run', () => {
+        generator.log.error = message => {
+          done(new Error(message));
+        };
+        generator.stdout = msg => {
+          const crd = JSON.parse(msg);
+          types[crd.type] = (types[crd.type] || 0) + 1;
+          if (crd.type === 'key-auth') {
+            keyAuthKeys.push(crd.keyId);
+          }
+        };
+      });
+
+      generator.once('end', () => {
+        keyAuthKeys.sort();
+        createdKeyAuthKeys.all.sort();
+
+        assert.deepEqual(types, createdTypes.active);
+        assert.deepEqual(keyAuthKeys, createdKeyAuthKeys.active);
+        done();
+      });
     });
+
+    env.argv = program.parse('credentials list -c ' + username);
   });
 
-  it('should show credentials', done => {
-    let output = {};
+  it('should show all deactivated credentials', done => {
+    const types = {};
+    const keyAuthKeys = [];
     env.hijack(namespace, generator => {
       generator.once('run', () => {
         generator.log.error = message => {
           done(new Error(message));
         };
         generator.stdout = msg => {
-          let crd = JSON.parse(msg);
-          output[crd.type] = (crd.keyId || crd.secret || crd.password);
+          const crd = JSON.parse(msg);
+          types[crd.type] = (types[crd.type] || 0) + 1;
```

Comment:
didn't get the magic with `|| 0 +1`

Reply:
try add 1 to undefined ... you will get NaN

Reply:
ok, I get it, you are trying to count number of repeats and after that put key-auth keys separately
as well as this sorting      `createdKeyAuthKeys.deactivated.sort();` and deep equal 

I can live with that. 

From my understanding there is simpler way 
this line `output[crd.type] = (crd.keyId || crd.secret || crd.password);`
could be changed to 
output[crd.type +  (crd.keyId || crd.secret || crd.password)] = true;

so it will be object with props:
"basic-authsomemagichere"
"key-authbase62guid" 

so no sorting, NaN check, all unique 

up to you

Reply:
The problem is that object returned from `adminHelper.admin.credentials.create` does not contain same fields as this one from cli. 

Example (`basic-auth` credential):

`adminHelper.admin.credentials.create`:
```json
{ 
  "isActive": true,
  "createdAt": "Wed Sep 27 2017 15:43:01 GMT+0200 (CEST)",
  "updatedAt": "Wed Sep 27 2017 15:43:01 GMT+0200 (CEST)",
  "id": "6qg7msNHt8xxNCjEnvoXUX"
}
```

CLI:
```json
{ 
  "createdAt": "Wed Sep 27 2017 15:43:01 GMT+0200 (CEST)",
  "isActive": true,
  "password": "$2a$10$mc38mMk1ZtU6qlJbsJ.O7.iCJiefQP3l.NVzr1gcWeqh6map6snLi",
  "updatedAt": "Wed Sep 27 2017 15:43:01 GMT+0200 (CEST)",
  "type": "basic-auth"
}
```

Reply:
ok, I don't mind against this part of the code. 

Important part that you found a bug or even 2. 
1) crd.type should always be filled, for create, get etc. operations
should be one line in  credential service (REST API for create) (Swagger reference promises "type" field everywhere)
 
2) and most likely there is not need for password hash in cli info output 
CLI needs password output when it autogenerates passwords, but after that hash should not be revealed 

I would prefer fix here, but feel free to register  tickets.

Reply:
Done #468